js/reference.js

```diff
@@ -2127,25 +2151,103 @@
 
                     // check url length limit if not first one;
                     if (i != 0 && new Reference(module.parse(baseUri + currentPath + (i != 0 ? ';' : '') + filter), self.table.schema.catalog).contextualize.compact._getReadPath().value.length > module.URL_PATH_LENGTH_LIMIT) {
```

Comment:
You don't need to call the Reference API at all. Just check the `baseUri + currentPath + (i != 0 ? ';' : '') + filter` length directly.

Reply:
This was taken from the function how you wrote it before. I see why it's unnecessary.